cocos2d/core/CCNode.js

```diff
@@ -508,8 +508,9 @@ function _searchMaskInParent (node) {
             }
         }
 
-        return list;
+        return list.next ? list : null;
```

Comment:
感觉这样会好点，可以吗？
```
        // Because Mask may be nested, need to find all the Mask components in the parent node. 
        // The click area must satisfy all Masks to trigger the click.
        let list = null;
        for (var curr = node; curr && cc.Node.isNode(curr); curr = curr._parent, ++index) {
            if (curr.getComponent(Mask)) {
                let next = {
                    index: index,
                    node: curr,
                    next: null
                };
                if (list) {
                    list.next = next;
                }
                else {
                    list = next;
                }
                list = list.next;
            }
        }
        return list;
```

Reply:
这里改为返回数组了，修改为通用的方法了。链表取消头部节点，后面循环做移除的时候还得记录上一个节点，直接改为返回数组吧。